frontend/packages/dev-console/src/components/pipelines/modals/common/ModalStructure.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikProps, FormikValues } from 'formik';
+import { Form } from '@patternfly/react-core';
+import { ModalBody, ModalSubmitFooter, ModalTitle } from '@console/internal/components/factory';
+
+type ModalStructureProps = FormikProps<FormikValues> & {
+  children: React.ReactNode;
+  close: () => void;
+  submitBtnText: string;
```

Comment:
Would it be a good idea to have `cancelBtnText` as well? This seems like a common component that can be used in other places as well.

Reply:
Perhaps, I'll log a refactor ticket for this. I think we need to focus on getting this in asap rather than improving it for future use-cases. There will be cleanup work needed anyways. Good suggestion though!

Reply:
https://issues.redhat.com/browse/ODC-3608